src/Solnet.Rpc/Utilities/AddressExtensions.cs

```diff
@@ -75,6 +78,7 @@ public static (byte[] Address, int Nonce) FindProgramAddress(IEnumerable<byte[]>
                 catch (Exception)
                 {
                     buffer.RemoveAt(buffer.Count - 1);
+                    nonce--;

```

Comment:
Is exception-driven logic the only solution here?

Reply:
Probably not, would require refactoring CreateProgramAddress and FindProgramAddress to return a `null` or something when it can't find an address that doesn't fall off the curve

Reply:
I would prefer the TrySomething approach
```
bool TryCreateProgramAddress(buffer, programId, out result);
```

Reply:
Going with TrySomething approach